xbmc/peripherals/devices/PeripheralCecAdapter.cpp

```diff
@@ -1185,12 +1185,12 @@ void CPeripheralCecAdapter::CecSourceActivated(void *cbParam, const CEC::cec_log
         // pause/resume player
         CApplicationMessenger::GetInstance().SendMsg(TMSG_MEDIA_PAUSE);
     }
-    else if (bPlayingAndDeactivated
-      && adapter->GetSettingInt("pause_or_stop_playback_on_deactivate") == LOCALISED_ID_STOP)
+    else if (activated == 0 && adapter->GetSettingInt("pause_or_stop_playback_on_deactivate") == LOCALISED_ID_STOP)
     {
       if (pSlideShow)
         pSlideShow->OnAction(CAction(ACTION_STOP));
-      else
+      else if (g_application.m_pPlayer->IsPlaying())
+        // only when playing as TMSG_MEDIA_STOP will wake the screensaver, this can make the source active again
```

Comment:
Perhaps a better solution would be to not wake the screen saver when stop is pressed? Most people just press any button I think, I doubt anyone would even notice if stop doesn't work.